src/events/event_bus.h

```diff
@@ -0,0 +1,77 @@
+//
+// Created by Piotr Moszkowicz on 2019-02-12.
+//
+
+#pragma once
+#include <list>
+#include <typeindex>
+
+#include "pch.h"
+#include "base_handler.h"
+#include "function_handler.h"
+#include "member_handler.h"
+#include "member_async_handler.h"
+
+
+namespace Bald::Events {
+    typedef std::list<BaseHandler*> HandlersList;
+
+    enum HandlerType {
+        SyncHandler = 0,
+        AsyncHandler = 1
+    };
+
+    class EventBus {
+    public:
+        template<typename EventType>
+        static void emit(EventType* e) noexcept {
+            HandlersList * handlers = m_subscribers[typeid(EventType)];
+
+            if (handlers == nullptr) return;
+
+            for (auto & handler : *handlers) {
+                if (handler != nullptr) {
+                    handler->exec(e);
+                }
+            }
```

Comment:
delete EventType after this function is executed -> we will allocate it dynamically in all scenerios